### ISSUE TITLE
grouping options not recognized in overrides

### ISSUE THREAD

**Issue by mamoll** [inukshuk/jekyll-scholar#129]
I amusing jekyll 3.1.1 with jekyll-scholar 5.6.0. In my `_config.yml` I have the following sorting/grouping-related settings:
```
    sort_by: year
    order: descending
    group_order: ascending
```
I'd expect the following snippet to generate a bibliography with groups in descending order, but they are still in ascending order:
```
{% bibliography --group_by year --group_order descending %}
```
Similarly, I'd expect the following snippet to group publications by type and list them in the order specified:
```
{% bibliography --group_by type --type_order [book,article,incollection,inproceedings,thesis,techreport] %}
```
They *are* grouped by type, but the type order is different.

BTW, it'd be really nice if there was some way to insert anchors for each group and generate a list of links to these anchors. This is my hacky way to do this (which involved manually publication types):
```
<div class="pubindex">
  <ul>
  {% for pubtype in site.data.pubtype %}<li><a href="#{{pubtype.type | remove: ' '}}">{{pubtype.type}}</a></li>{% endfor %}
  </ul>
</div>

{% for pubtype in site.data.pubtype %}

<h2><a name="{{pubtype.type | remove: ' '}}"></a>{{pubtype.type}}</h2>

{% bibliography --query {{pubtype.query}} %}

{% endfor %}
```

**Comment by inukshuk** [comment]
I think we're looking only for `group_by` overrides so far; would make sense to override `group_order` too. Do we actually have `type_order` or is this a suggestion for something new? (Losing track of what has been added recently!)
 
cc @hendrikvanantwerpen @pkok

**Comment by mamoll** [comment]
Yes, `type_order` exists. If add this to `_config.yml`:
```
    group_by: type
    group_order: ascending
    type_order: [book,article,incollection,inproceedings,thesis,techreport]
```
then `{% bibliography %}` just works as expected.

**Comment by inukshuk** [comment]
Cool, so it's just a matter of overriding those two options. Should be very easy.

**Comment by inukshuk** [comment]
Landed in 5.7.0
